### PR TITLE
Add support for the new Data Browser API

### PR DESCRIPTION
**Pull request by luispuig**
React router launched the new Data Browser with a structure very similar to Remix with a loader, action, errorBoundary and handle options. This PR shows a possible way to add support for it

On the other hand, with the new browser API, part of the options/code could be removed:

* The 404 route. The new convention uses the ErrorbOundary for all the errors, including the 404.
* Manual routes or initializing with `root: { path: "", id: "root", file: "routes/index" }`. The new API allows adding them easily. This is an example of adding the root layout:
```javascript
import { createBrowserRouter, RouterProvider } from 'react-router-dom';
import remixRoutes from 'virtual:remix-routes';
import Root from './root';

const routes = createBrowserRouter([
  {
    path: '/',
    element: <Root />,
    children: remixRoutes,
  },
]);

const App = () => {
  return (
      <RouterProvider router={routes} fallbackElement={<div>loading</div>} />
  );
};

```

Finally, I noticed that the routes are not re-generated on reloading. I have solved it using the plugin `vite-plugin-full-reload`, it could be interesting adding it to the documentation.

### CODE REVIEW

File: lib/node/utils.ts [hunk 1]
```diff
@@ -68,6 +68,19 @@ function routeToString(
...
 
+  if (importMode === "dataRoute") {
+    components.sync.push(
+      `import * as ${componentName} from '${componentPath}';`
+    );
+
+    props.set("element", `createElement(${componentName}.default)`);
+    props.set("loader", `${componentName}.loader`);
+    props.set("action", `${componentName}.action`);
+    props.set("errorElement", `${componentName}.errorBoundary`);
```
Comment (luispuig) [inline lib/node/utils.ts]: I think that this one is wrong and it should be 
```
props.set(
      'errorElement',
      `${componentName}.ErrorBoundary !== undefined ? createElement(${componentName}.ErrorBoundary) : undefined`,
    );
```

File: lib/node/utils.ts [hunk 2]
```diff
@@ -68,6 +68,19 @@ function routeToString(
     props.set("element", `createElement(${componentName})`);
   }
 
+  if (importMode === "dataRoute") {
+    components.sync.push(
+      `import * as ${componentName} from '${componentPath}';`
+    );
+
+    props.set("element", `createElement(${componentName}.default)`);
```
Comment (luispuig) [inline lib/node/utils.ts]: I'd also do something here for resources routes (without default component)
```
    props.set(
      'element',
      `${componentName}.default !== undefined ? createElement(${componentName}.default) : undefined`,
    );
```

**Comment by vjee** [comment]
Hi Luis, Thanks again for this PR!

I have added a commit to it to get it ready for merge.
As far as I can see this one should be good to go.

Would you like to confirm if it's working as you expect before I merge?

**Comment by luispuig** [comment]
Looks good to me. Thanks for the effort.

**Pull request merged**